python/pyarrow/table.pxi

```diff
@@ -189,6 +185,12 @@ cdef class ChunkedArray(_PandasConvertible):
         """
         return _pc().is_valid(self)
 
+    def __eq__(self, other):
+        try:
+            return self.equals(other)
+        except TypeError:
+            return NotImplemented
```

Comment:
I doubt a bit whether we should add this back (this is from the revert, since ChunkedArray implemented this, while Array did not). Or otherwise we should maybe rather add it like this to Array as well ..

Right now, RecordBatch and Table also have a `__eq__` that dispatches to `equals()`.

Reply:
I think adding it to Array for consistency sounds good

Reply:
OK, updated Array